assets/js/hash-worker.js

```diff
@@ -1,75 +1,26 @@
-/* jslint browser: true, indent: 4 */
-/* global self, require, postMessage, asmCrypto, console, FileReaderSync */
+/* global self, require, postMessage, asmCrypto, FileReaderSync */
 
-require('asmcrypto.js');
+require('asmcrypto.js'); // TODO: import
 
-var blockSize = 1048576;
+const BLOCK_SIZE = 1048576;
 
-self.addEventListener('message', function(evt) {
-    // Unpack some variables for clarity below:
-    var d = evt.data,
-        workerId = d.workerId,
-        action = d.action,
-        file = d.file,
-        fullPath = d.fullPath,
-        response = {'file': file, 'fullPath': fullPath, 'action': action, 'workerId': workerId};
+self.addEventListener('message', ({data: {file, fullPath}}) => {
+    const sha256 = new asmCrypto.SHA256();
```

Comment:
My main argument for adding it would be to ensure that our code handles n > 1 hasher running in the future by keeping it working with lists even if there's only one element.